File: ipman.shared/Entity/UserAccount.cs
using ipman.shared.Entity.Join;
using ipman.shared.Entity.Lookups;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;

namespace ipman.shared.Entity
{
   public  class UserAccount: EntityBase
   {
        public string EmailAddress { get; set; }
        public string Username { get; set; }
        public string UserAccountSalt { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string GitHubID { get; set; }
        public string GoogleID { get; set; }
        public string AvatarLink { get; set; }
        public bool IsActive { get; set; } = true;
        public AuthenticationProvider LastLoginProvider { get; set; }
        public DateTime LastLoginUTC { get; set; }
        public DateTime CreatedUTC { get; set; }
        public DateTime LastUpdatedUTC { get; set; }
        public virtual ICollection<Vote> Votes { get; set; }
        public virtual ICollection<Post> CreatedPosts { get; set; }
        public virtual ICollection<SiteAccountUserAccount> SiteAccountUserAccounts { get; set; }
        [NotMapped]
        public ICollection<SiteAccount> SiteAccounts => SiteAccountUserAccounts?.Select(saua => saua.SiteAccount)?.ToList();
    }
}

File: ipman/ClientApp/entity/UserAccount.ts
import { EntityBase } from "./EntityBase";
import SiteAccount from "./SiteAccount";
import { AuthenticationProvider } from "./lookups/AuthenticationProvider";
import SiteAccountUserAccount from "./join/SiteAccountUserAccount";
import Post from "./Post";
import Vote from "./Vote";
import Wager from "./Wager";

export default interface UserAccount extends EntityBase {
    EmailAddress: string;
    Username: string;
    FirstName: string;
    LastName: string;
    GitHubID: string;
    GoogleID: string;
    AvatarLink: string;
    IsActive: boolean;
    LastLoginProvider: AuthenticationProvider;
    LastLoginUTC: Date | string;
    CreatedUTC: Date | string;
    LastUpdatedUTC: Date | string;
    Votes: Vote[];
    Wagers: Wager[];
    CreatedPosts: Post[];
    SiteAccountUserAccounts: SiteAccountUserAccount[];
    SiteAccounts: SiteAccount[];
}
